pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -1757,13 +1743,22 @@ func (b *changefeedResumer) resumeWithRetries(
 
 			confPoller := make(chan struct{})
 			g := ctxgroup.WithContext(ctx)
-			targets, err := AllTargets(ctx, details, execCfg)
+			initialHighWater, schemaTS, err := computeDistChangefeedTimestamps(ctx, jobExec, details, localState)
+			if err != nil {
+				return err
+			}
+			targets, err := AllTargets(ctx, details, execCfg, schemaTS)
 			if err != nil {
 				return err
 			}
+			maybeCfKnobs, haveKnobs := execCfg.DistSQLSrv.TestingKnobs.Changefeed.(*TestingKnobs)
+			if haveKnobs && maybeCfKnobs.TablesetChangingCallback != nil {
```

Comment:
`AllTargets` is called in multiple places, could you explain why this site was picked for `TablesetChangingCallback`?